File: Algorithm/c_or_cpp/charpter13_digital/level3/is_prime.cpp
#include <stdio.h>
#include <math.h>

int isPrime(int num) {
    int max = (int) sqrt(num);
    for (int i = 2; i <= max; i++) {
        if (num % i == 0) {
            return 0;
        }
    }
    return 1;
}

int main() {
    int num;
    printf("Enter a number: ");
    scanf("%d", &num);
    if (isPrime(num)) {
        printf("%d is a prime number.", num);
    } else {
        printf("%d is not a prime number.", num);
    }
    return 0;
}

File: Algorithm/Code_Materials/Java/src/main/java/com/yugutou/charpter13_math/level3/IsPrime.java
package com.yugutou.charpter13_math.level3;

public class IsPrime {
    public static void main(String[] args) {

        System.out.println(isPrime(11));
    }

   public static boolean isPrime(int num) {
        int max = (int)Math.sqrt(num);
        for (int i = 2; i <= max; i++) {
            if (num % i == 0) {
                return false;
            }
        }
        return true;
    }
}
